python/pyspark/mlv2/classification.py

```diff
@@ -0,0 +1,309 @@
+
+        num_batches_per_worker = math.ceil(num_rows / num_train_workers / batch_size)
+        num_samples_per_worker = num_batches_per_worker * batch_size
+
+        num_features = len(dataset.select(self.getFeaturesCol()).head()[0])
+
+        if num_classes < 2:
+            raise ValueError("Training dataset distinct labels must >= 2.")
+
+        # TODO: support GPU.
+        distributor = TorchDistributor(
+            local_mode=False, use_gpu=False, num_processes=num_train_workers
+        )
+        model_state_dict = distributor._train_on_dataframe(
+            _train_logistic_regression_model_worker_fn,
+            dataset,
+            num_samples_per_worker=num_samples_per_worker,
+            num_features=num_features,
+            batch_size=batch_size,
+            max_iter=self.getMaxIter(),
+            num_classes=num_classes,
+            learning_rate=self.getLearningRate(),
+            momentum=self.getMomentum(),
+            fit_intercept=self.getFitIntercept(),
+            seed=self.getSeed(),
+        )
+
+        dataset.unpersist()
+
+        torch_model = _LinearNet(
+            num_features=num_features, num_classes=num_classes, bias=self.getFitIntercept()
+        )
+        torch_model.load_state_dict(model_state_dict)
+
+        lor_model = LogisticRegressionModel(
+            torch_model, num_features=num_features, num_classes=num_classes
+        )
+        lor_model._resetUid(self.uid)
+        return self._copyValues(lor_model)
+
+
+@inherit_doc
+class LogisticRegressionModel(PredictionModel, _LogisticRegressionParams):
+    """
+    Model fitted by LogisticRegression.
+
+    .. versionadded:: 3.5.0
+    """
+
+    def __init__(self, torch_model, num_features, num_classes):
+        super().__init__()
+        self.torch_model = torch_model
+        self.num_features = num_features
+        self.num_classes = num_classes
+
+    def numFeatures(self) -> int:
+        return self.num_features
+
+    def numClasses(self) -> int:
+        return self.num_classes
+
+    def _input_column_name(self) -> str:
+        return self.getOrDefault(self.featuresCol)
+
+    def _output_columns(self) -> List[Tuple[str, str]]:
+        output_cols = [(self.getOrDefault(self.predictionCol), "bigint")]
+        prob_col = self.getOrDefault(self.probabilityCol)
+        if prob_col:
+            output_cols += [(prob_col, "array<double>")]
+        return output_cols
+
+    def _get_transform_fn(self) -> Callable[["pd.Series"], Any]:
+        model_state_dict = self.torch_model.state_dict()
+        num_features = self.num_features
+        num_classes = self.num_classes
+        fit_intercept = self.getFitIntercept()
+
+        def transform_fn(input_series: "pd.Series") -> Any:
+            torch_model = _LinearNet(
+                num_features=num_features, num_classes=num_classes, bias=fit_intercept
+            )
+            # TODO: Use spark broadast for `model_state_dict`,
```

Comment:
I guess we don't have plan to support broadcast

Reply:
OK. Not a big issue for now.